src/theory/booleans/theory_bool.cpp

```diff
@@ -40,20 +40,20 @@ Theory::PPAssertStatus TheoryBool::ppAssert(TNode in, SubstitutionMap& outSubsti
 
   // Add the substitution from the variable to its value
   if (in.getKind() == kind::NOT) {
-    if (in[0].getKind() == kind::VARIABLE) {
+    if (in[0].isVar())
+    {
       outSubstitutions.addSubstitution(in[0], NodeManager::currentNM()->mkConst<bool>(false));
-    } else {
-      return PP_ASSERT_STATUS_UNSOLVED;
+      return PP_ASSERT_STATUS_SOLVED;
```

Comment:
Do we not have to return PP_ASSERT_STATUS_UNSOLVED in the else case here?

Reply:
The previous code was returning PP_ASSERT_STATUS_UNSOLVED when in/in[0] *is not* a variable, and fell through to return PP_ASSERT_STATUS_SOLVED.  Since the new code makes it so that there is another case to handle (equalities between variables), the code was flipped so that we return PP_ASSERT_STATUS_SOLVED if in/in[0] *is* a variable, and fall through to call the default method otherwise.